lib/grp.gi

```diff
@@ -176,9 +176,13 @@ InstallMethod( IsPGroup,
         if ord = infinity then
           return false;
         elif 1 < ord then
-          UniteSet( s, Factors( ord ) );
-          if 1 < Length( s ) then
+          if not IsPrimePowerInt( s ) then
```

Comment:
Yes, sorry. I made this typo at two places, and found only one before....

I will try to look at the structure of these tst files and add one for p-groups.